src/libcprover-cpp/api.cpp

```diff
@@ -47,8 +42,58 @@ api_sessiont::api_sessiont(const api_optionst &options)
 
 api_sessiont::~api_sessiont() = default;
 
-void api_sessiont::load_model_from_files(const std::vector<std::string> &files)
+struct api_messaget
 {
+  std::string string;
+};
+
+const char *api_message_get_string(const api_messaget &message)
+{
+  return message.string.c_str();
+}
+
+class api_message_handlert : public message_handlert
+{
+public:
+  explicit api_message_handlert(
+    api_message_callbackt callback,
+    api_call_back_contextt context);
+
+  void print(unsigned level, const std::string &message) override;
+
+  // Unimplemented for now. We may need to implement these as string conversions
+  // or something later.
+  void print(unsigned level, const xmlt &xml) override{};
+  void print(unsigned level, const jsont &json) override{};
+
+  void flush(unsigned) override{};
+
+private:
+  api_call_back_contextt context;
+  api_message_callbackt callback;
+};
+
+api_message_handlert::api_message_handlert(
+  api_message_callbackt callback,
+  api_call_back_contextt context)
+  : message_handlert{}, context{context}, callback{callback}
+{
+}
+
+void api_message_handlert::print(unsigned level, const std::string &message)
+{
+  if(!callback)
```

Comment:
Do we want to allow `null` callbacks? What's the usecase?

Reply:
We use a `null_message_handlert` within the cbmc code base as part of our unit test suite where it is desirable to discard the status messages rather than print them to the terminal. It can also be handy for local prototyping purposes, as it means there is one less thing to setup.

Just not printing in the case of a null pointer seems like the preferential behaviour to me at the moment. The alternatives that currently come to mind would terminate the process.